runelite-client/src/main/java/net/runelite/client/plugins/inventoryviewer/InventoryViewerOverlay.java

```diff
@@ -77,24 +86,42 @@ public Dimension render(Graphics2D graphics)
 
 		final Item[] items = itemContainer.getItems();
 
-		for (int i = 0; i < INVENTORY_SIZE; i++)
+		if (config.viewerMode() == InventoryViewerMode.FULL)
 		{
-			if (i < items.length)
+			for (int i = 0; i < INVENTORY_SIZE; i++)
 			{
-				final Item item = items[i];
-				if (item.getQuantity() > 0)
+				if (i < items.length)
 				{
-					final BufferedImage image = getImage(item);
-					if (image != null)
+					final Item item = items[i];
+					if (item.getQuantity() > 0)
 					{
-						panelComponent.getChildren().add(new ImageComponent(image));
-						continue;
+						final BufferedImage image = getImage(item);
+						if (image != null)
+						{
+							panelComponent.getChildren().add(new ImageComponent(image));
+							continue;
+						}
 					}
 				}
+
+				// put a placeholder image so each item is aligned properly and the panel is not resized
+				panelComponent.getChildren().add(PLACEHOLDER_IMAGE);
 			}
+		}
+		else
+		{
+			final Map<Integer, Integer> totals = Arrays.stream(items)
```

Comment:
This else block can be in an if statement before the above block with early return